### PR TITLE
Added steps to create github releases

### PR DESCRIPTION
**Pull request by levius0**
Added several steps to create a github release every time we merge to main.

### CODE REVIEW

**Review by ehsan-fj — APPROVED** [review]

**Review by ehsan-fj — COMMENTED** [review]
@levius0 where do you attach the assets?

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -49,6 +49,54 @@ jobs:
...
+                  "title": "## 📦 Dependencies",
+                  "labels": ["dependencies"]
+              }
+            ]
+          }
+        ignorePreReleases: "false"
+        fromTag: ${{ steps.previous_tag.outputs.tag }}
+        toTag: ${{env.NEW_VERSION}}
+        token: ${{ secrets.GITHUB_TOKEN }}
+    - name: Create release
```
Comment (levius0) [inline .github/workflows/deployment.yml:91]: This step automatically attach repo as asset in zip and tar.

**Comment by levius0** [comment]
> @levius0 where do you attach the assets?

Create release step automatically attaches assets as zip and tar.gz.

**Review by ehsan-fj — APPROVED** [review]
Great job 🚀

**Pull request merged**